gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/VerificationOnboardService.java

```diff
@@ -74,58 +70,114 @@ public Optional<String> findSwaggerUrl(Map<String, String> metadata) {
+
+        HttpHeaders headersNoSSO = new HttpHeaders();
+        headersNoSSO.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<String> requestNoSSO = new HttpEntity<>(headersNoSSO);
+
         for (Endpoint endpoint : getEndpoints) {
-            String urlFromSwagger = endpoint.getUrl();
-            // replaces parameters in {} in query
-            String url = urlFromSwagger.replaceAll("\\{[^{}]*}", "dummy");
-
-            ResponseEntity<String> response;
-            try {
-                response = restTemplate.getForEntity(url, String.class);
-            } catch (HttpClientErrorException | HttpServerErrorException e) {
-                response = ResponseEntity.status(e.getRawStatusCode())
-                    .headers(e.getResponseHeaders())
-                    .body(e.getResponseBodyAsString());
+            for (HttpMethod method : endpoint.getHttpMethods()) {
+                checkEndpoint(endpoint, result, method, requestSSO, requestNoSSO);
             }
+        }
+        return result;
+    }
+
+    private void checkEndpoint(Endpoint endpoint, ArrayList<String> result, HttpMethod method, HttpEntity<String> requestSSO, HttpEntity<String> requestNoSSO) {
+        if (method == HttpMethod.DELETE) {
+            return;
+        }
 
-            String responseBody = response.getBody();
+        String urlFromSwagger = endpoint.getUrl();
+        // replaces parameters in {} in query
+        String url = urlFromSwagger.replaceAll("\\{[^{}]*}", "dummy");
 
-            if (responseBody != null && response.getStatusCode() == HttpStatus.NOT_FOUND && responseBody.contains("ZWEAM104E")) {
-                result.add("Documented endpoint at " + endpoint.getUrl() + " could not be located, attempting to call it through gateway gives the ZWEAM104E error");
-            }
+        ResponseEntity<String> responseWithSSO = getResponse(requestSSO, method, url);
+        result.addAll(fromResponseReturnFoundProblems(responseWithSSO, endpoint, method, true));
 
-            if (!endpoint.isGetResponseCodeDocumented(String.valueOf(response.getStatusCode().value()))) {
-                result.add("Calling endpoint at " + endpoint.getUrl() + " gives undocumented " + response.getStatusCode().value()
-                    + " status code, documented responses are:" + endpoint.getValidResponses().get("GET"));
-            }
+        ResponseEntity<String> responseNoSSO = getResponse(requestNoSSO, method, url);
+        result.addAll(fromResponseReturnFoundProblems(responseNoSSO, endpoint, method, false));
+    }
+
+    private ResponseEntity<String> getResponse(HttpEntity<String> request, HttpMethod method, String url) {
+        ResponseEntity<String> response;
+        try {
+            response = restTemplate.exchange(url, method, request, String.class);
+        } catch (HttpClientErrorException | HttpServerErrorException e) {
+            response = ResponseEntity.status(e.getRawStatusCode())
```

Comment:
in this case it is better to use Apache HTTP client since that one is not throwing exceptions for unauthorized, internal error,.. codes

Reply:
It would require lot of changes